server/datastore/mysql/microsoft_mdm.go

```diff
@@ -101,7 +78,11 @@ func (ds *Datastore) MDMWindowsInsertEnrolledDevice(ctx context.Context, device
 	return nil
 }
 
-// MDMWindowsDeleteEnrolledDevice deletes a give MDMWindowsEnrolledDevice entry from the database using the device id.
+// TODO(mna): should we have something like host_dep_assignments for Windows? I don't remember exactly what
+// problem this was solving, but seeing those enrollments deletion made me think of it.
```

Comment:
`host_dep_assignments` is meant to keep track of devices assigned to us in ABM, they are in a limbo as they're not yet enrolled (so we can't track them in the nano tables)

I'm not sure if we need to do the same for Microsoft Autopilot, but sounds like something out of this scope (thankfully)